static/js/map/map.routes.js

```diff
@@ -0,0 +1,252 @@
+    }
+
+    return marker
+}
+
+function getRouteColor(route) {
+    const colorTable = ['#DA051B',
+                        '#E84921',
+                        '#EF7D1D',
+                        '#F8B310',
+                        '#F3E500',
+                        '#8EB71B',
+                        '#229548',
+                        '#0094AA',
+                        '#1F4995',
+                        '#172C85',
+                        '#4F2577',
+                        '#A0077C',
+                        '#F5B3F9']
+
+    if (route.route_id) {
+        return colorTable[parseInt(route.route_id.slice(-5,-3),16)%colorTable.length]
+    }
+    return colorTable[Math.floor(Math.random() * colorTable.length)]
+}
+
+function addPopup(route, marker) {
+    const popupContent = routeLabel(route, null)
+    marker.bindPopup(popupContent, { autoPan: autoPanPopup() })
+
+    marker.on('mouseover', function (e) {
+        this.openPopup(e.latlng)
+        this.setStyle({
+            weight: 6
+        })
+    })
+    marker.on('mouseout', function (e) {
+        this.closePopup()
+        this.setStyle({
+            weight: 3
+        })
+    })
+    marker.on('mousemove', function (e) {
+        this.closePopup()
+        this.openPopup(map.mouseEventToLatLng(e.originalEvent))
+        this.setStyle({
+            weight: 6
+        })
+    })
+}
+
+function setupRoutePath(route) {
+    let pointLL = new L.latLng(route.start_poi_latitude, route.start_poi_longitude)
+    let routePoints = [pointLL]
+
+    const wp = JSON.parse(route.waypoints)
+    for (let i = 0; i < wp.length; i++) {
+        if (!wp[i].lat_degrees || !wp[i].lng_degrees) {
+            console.log('Unknown route waypoint: ', JSON.stringify(wp[i]))
+            continue
+        }
+        pointLL = new L.latLng(wp[i].lat_degrees, wp[i].lng_degrees)
+        routePoints.push(pointLL)
+    }
+    pointLL = new L.latLng(route.end_poi_latitude, route.end_poi_longitude)
+    routePoints.push(pointLL)
+
+    L.ClusterablePolyline = L.Polyline.extend({
+        _originalInitialize: L.Polyline.prototype.initialize,
+
+        initialize: function (bounds, options) {
+            this._originalInitialize(bounds, options)
+            this._latlng = this.getBounds().getCenter()
+        },
+
+        //getLatLng: function () {
+            //return this._latlng
+        //},
+
+        // dummy method.
+        setLatLng: function () {
+        }
+    })
+
+    const routePath = new L.ClusterablePolyline(routePoints, {
+        color: getRouteColor(route),
+        weight: 3,
+        smoothFactor: 1
+    })
+    routePath.route_id = route.route_id
+    addPopup(route, routePath)
+    markers.addLayer(routePath)
+
+    return routePath
+}
+
+// left pad a string/input to a fixed length with extra padString
+function zpad(input, length, padString) {
+  let s = input.toString()
+  while (s.length < length) {
+    s = padString + s
+  }
+  return s
+}
```

Comment:
I guess you could use the already existing padStart function

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

Reply:
RocketMAD already had an lpad function, I'll use that.